Submitted By: Matthew Burgess <matthew@linuxfromscratch.org>
Date: 2005-07-15
Initial Package Version: 2.3.5
Upstream Status: From Upstream
Origin: Ryan Oliver, courtesy of Dan Kegel's crosstool
        (http://kegel.com/crosstool/)
Description: Fixes a compilation problem with gcc-4.0.x that results in:
             jis0208.h:32: error: array type has incomplete element type

diff -Naur glibc-2.3.5.orig/iconvdata/jis0208.h glibc-2.3.5/iconvdata/jis0208.h
--- glibc-2.3.5.orig/iconvdata/jis0208.h	2003-06-11 21:40:42.000000000 +0000
+++ glibc-2.3.5/iconvdata/jis0208.h	2005-07-15 18:32:20.726328528 +0000
@@ -24,15 +24,6 @@
 #include <gconv.h>
 #include <stdint.h>
 
-/* Conversion table.  */
-extern const uint16_t __jis0208_to_ucs[];
-
-extern const char __jisx0208_from_ucs4_lat1[256][2];
-extern const char __jisx0208_from_ucs4_greek[0xc1][2];
-extern const struct jisx0208_ucs_idx __jisx0208_from_ucs_idx[];
-extern const char __jisx0208_from_ucs_tab[][2];
-
-
 /* Struct for table with indeces in UCS mapping table.  */
 struct jisx0208_ucs_idx
 {
@@ -42,6 +33,15 @@
 };
 
 
+/* Conversion table.  */
+extern const uint16_t __jis0208_to_ucs[];
+
+extern const char __jisx0208_from_ucs4_lat1[256][2];
+extern const char __jisx0208_from_ucs4_greek[0xc1][2];
+extern const struct jisx0208_ucs_idx __jisx0208_from_ucs_idx[];
+extern const char __jisx0208_from_ucs_tab[][2];
+
+
 static inline uint32_t
 __attribute ((always_inline))
 jisx0208_to_ucs4 (const unsigned char **s, size_t avail, unsigned char offset)
